brave/src/main/java/com/linecorp/decaton/processor/runtime/BraveTracingProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.runtime;
+
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import brave.Span;
+import brave.kafka.clients.KafkaTracing;
+import brave.propagation.CurrentTraceContext.Scope;
+
+public class BraveTracingProvider implements TracingProvider {
+    private static final ThreadLocal<Scope> CURRENT_SCOPE = new ThreadLocal<>();
+    private final KafkaTracing kafkaTracing;
+
+    public BraveTracingProvider(KafkaTracing kafkaTracing) {
+        this.kafkaTracing = kafkaTracing;
+    }
+
+    @Override
+    public TraceHandle traceFor(ConsumerRecord<?, ?> record) {
+        final Span span = kafkaTracing.nextSpan(record).name("decaton").start();
+        return new TraceHandle() {
```

Comment:
For this point: https://github.com/line/decaton/pull/69/files#r500168178

Developers might cast `ProcessingContext#trace()` to specific TraceHandle to use impl-specific information (like brave's `Span` (though currently Span isn't exposed and I think it's not necessary to expose for now)), so shouldn't we define something like `BraveTraceHandle` rather than instantiating anonymous class ?

Reply:
If we don't expose the `Span` then I don't think it makes any sense to create a specific class either, since there's nothing a user can actually do with that `BraveTraceHandle`? My instinct is to keep everything hidden unless there's a use case for exposing the impl-specific implementation, but I'm happy to expose it if you think that's better.

Reply:
I see.
Regarding this: https://github.com/line/decaton/pull/69#discussion_r502987749 , we don't provide a way to access underlying trace for now so anonymous instantiation should be fine.